presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -180,7 +183,20 @@ public ConnectorSplitSource getSplits(ConnectorSession session, ConnectorTableLa
                 recursiveDfsWalkerEnabled);
 
         HiveSplitSource splitSource = new HiveSplitSource(connectorId, maxOutstandingSplits, hiveSplitLoader, executor);
-        hiveSplitLoader.start(splitSource);
+
+        UserGroupInformation ugi = UserGroupInformation.createRemoteUser(session.getUser());
+
+        try {
+            ugi.doAs((PrivilegedExceptionAction<Void>)
+                () -> {
```

Comment:
Looks like this needs to be right shifted. Presto follows this style:
https://github.com/airlift/codestyle